tfhe/src/integer/server_key/radix_parallel/modulus_switch_compression.rs

```diff
@@ -61,12 +51,97 @@ impl ServerKey {
         &self,
         compressed_ct: &CompressedModulusSwitchedSignedRadixCiphertext,
     ) -> SignedRadixCiphertext {
-        let blocks = compressed_ct
-            .blocks
+        BaseSignedRadixCiphertext {
+            blocks: self.decompress_generic_parallelized(&compressed_ct.0),
+        }
+    }
+
+    #[allow(clippy::int_plus_one)]
+    fn switch_modulus_and_compress_generic_parallelized(
+        &self,
+        blocks: &[Ciphertext],
+    ) -> CompressedModulusSwitchedRadixCiphertextGeneric {
+        assert!(
+            self.message_modulus().0 <= self.carry_modulus().0,
+            "Compression does not support message_modulus > carry_modulus"
+        );
+        assert!(
+            self.key.max_noise_level.get() >= self.message_modulus().0 + 1,
+            "Compression does not support max_noise_level < message_modulus + 1"
+        );
+
+        let paired_blocks;
+
+        let last_block;
+
+        let len = blocks.len();
+
+        if len % 2 == 0 {
+            paired_blocks = blocks;
+            last_block = None;
+        } else {
+            paired_blocks = &blocks[..len - 1];
+            last_block = Some(blocks.last().unwrap());
+        }
+
+        let paired_blocks = paired_blocks
             .par_iter()
-            .map(|a| self.key.decompress(a))
+            .chunks(2)
+            .map(|pair| {
+                assert_eq!(pair.len(), 2);
```

Comment:
is this really required ?

Reply:
Replacing by `.par_iter().chunks(2)` by `.par_chunks_exact(2)`